File: javascript/Two Sum.ts
function twoSum(nums: number[], target: number): number[] {
    const seen = new Map<number, number>(); // value -> index

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (seen.has(complement)) {
            return [seen.get(complement)!, i];
        }
        seen.set(nums[i], i);
    }

    return []; // في حال مفيش حل (لكن السؤال ضامن وجود حل)
}

File: python/Two Sum.py
# https://leetcode.com/problems/two-sum/description/


class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        num_list = range(len(nums))
        for n in num_list : 
            for i in num_list[n+1:] :
                if nums[n] + nums[i] == target :
                    return [i, n]
            
